localstack/services/sns/resource_providers/aws_sns_topic.py

```diff
@@ -0,0 +1,185 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, Type, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    CloudFormationResourceProviderPlugin,
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+from localstack.utils.strings import short_uid
+
+
+class SNSTopicProperties(TypedDict):
+    ContentBasedDeduplication: Optional[bool]
+    DataProtectionPolicy: Optional[dict]
+    DisplayName: Optional[str]
+    FifoTopic: Optional[bool]
+    KmsMasterKeyId: Optional[str]
+    SignatureVersion: Optional[str]
+    Subscription: Optional[list[Subscription]]
+    Tags: Optional[list[Tag]]
+    TopicArn: Optional[str]
+    TopicName: Optional[str]
+    TracingConfig: Optional[str]
+
+
+class Subscription(TypedDict):
+    Endpoint: Optional[str]
+    Protocol: Optional[str]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class SNSTopicProvider(ResourceProvider[SNSTopicProperties]):
+    TYPE = "AWS::SNS::Topic"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[SNSTopicProperties],
+    ) -> ProgressEvent[SNSTopicProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/TopicArn
+
+
+
+        Create-only properties:
+          - /properties/TopicName
+          - /properties/FifoTopic
+
+        Read-only properties:
+          - /properties/TopicArn
+
+        IAM permissions required:
+          - sns:CreateTopic
+          - sns:TagResource
+          - sns:Subscribe
+          - sns:GetTopicAttributes
+          - sns:PutDataProtectionPolicy
+
+        """
+        model = request.desired_state
+        sns = request.aws_client_factory.sns
+        # TODO: validations and iam checks
+
+        attributes = {k: v for k, v in model.items() if v is not None if k != "TopicName"}
+
+        # following attributes need to be str instead of bool for boto to work
+        if attributes.get("FifoTopic") is not None:
+            attributes["FifoTopic"] = str(attributes.get("FifoTopic"))
+
+        if attributes.get("ContentBasedDeduplication") is not None:
+            attributes["ContentBasedDeduplication"] = str(
+                attributes.get("ContentBasedDeduplication")
+            )
+
+        subscriptions = []
+        if attributes.get("Subscription") is not None:
+            subscriptions = attributes["Subscription"]
+            del attributes["Subscription"]
+
+        tags = []
+        if attributes.get("Tags") is not None:
+            subscriptions = attributes["Tags"]
```

Comment:
```suggestion
            tags = attributes["Tags"]
```

I think this is the actual cause of the KeyError from earlier. 
I was wondering why it even entered the for loop further below and it seems the tags are set as the subscriptions here (probably unintentionally) :grimacing: 

Removing this should also remove the need for the `if "Protocol" in subscription:`,

Reply:
good catch!